google-cloud-datastore/lib/google/cloud/datastore.rb

```diff
@@ -103,12 +105,13 @@ def self.new project_id: nil,
                    timeout: nil,
                    endpoint: nil,
                    emulator_host: nil,
+                   database_id: configure.database_id,
```

Comment:
I would set the default to nil, and then add a `database ||= configure.database_id` line below, like the other arguments. It makes it less error-prone, in case someone explicitly passes nil into the parameter.

Reply:
Cyclomatic and perceived complexity was too high with the recommended solution. I'll take a look and try to reduce it by some other changes.